.github/workflows/release-pull-request.yml

```diff
@@ -0,0 +1,94 @@
+name: Release & Publish Package
+
+on:
+  pull_request:
+    branches: [develop]
+
+  workflow_dispatch:
+
+jobs:
+  fork-check:
+    runs-on: ubuntu-latest
+    outputs:
+      is-not-fork: ${{ steps.check.outputs.is_not_fork }}
+    steps:
+      - id: check
+        run: echo "::set-output name=is_not_fork::${{ github.repository == 'secretkeylabs/xverse-core' }}"
+
+  test:
+    needs: fork-check
+    if: needs.fork-check.outputs.is-not-fork == 'true'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+      - run: npm ci
+      # TODO: enable linting once ready
+      # - run: npm run lint
+      - run: npm test
+
+  publish-beta:
+    needs: test
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+      pull-requests: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+          registry-url: https://npm.pkg.github.com/
+
+      - id: git-commit
+        run: echo "sha=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - run: npm ci
+      - run: npm run build:esm
+
+      - id: current-version
+        run: echo "version=$(npm pkg get version  | tr -d '"')" >> $GITHUB_OUTPUT
+      - id: tag-version
+        run: npm version --no-git-tag-version $CURRENT_VERSION-$SHA
+        env:
+          SHA: ${{ steps.git-commit.outputs.sha }}
+          CURRENT_VERSION: ${{ steps.current-version.outputs.version }}
+
+      - id: publish
+        run: npm publish
```

Comment:
We should probably publish this to a "pr" tag so that it doesn't become the "latest" version